server/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodesStatsRequestParametersTests.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.node.stats;
+
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.action.admin.cluster.node.stats.NodesStatsRequestParameters.Metric;
+import org.elasticsearch.common.io.stream.ByteArrayStreamInput;
+import org.elasticsearch.common.io.stream.BytesRefStreamOutput;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.EnumSet;
+import java.util.List;
+
+public class NodesStatsRequestParametersTests extends ESTestCase {
+
+    public void testReadWriteMetricSet() {
+        for (var version : List.of(TransportVersions.VERSIONED_MASTER_NODE_REQUESTS, TransportVersions.NODES_STATS_ENUM_SET)) {
+            var randSet = randomSubsetOf(Metric.ALL);
+            var metricsOut = randSet.isEmpty() ? EnumSet.noneOf(Metric.class) : EnumSet.copyOf(randSet);
+            try {
+                var out = new BytesRefStreamOutput();
+                out.setTransportVersion(version);
+                Metric.writeSetTo(out, metricsOut);
+                var in = new ByteArrayStreamInput(out.get().bytes);
+                in.setTransportVersion(version);
+                var metricsIn = Metric.readSetFrom(in);
+                assertEquals(metricsOut, metricsIn);
+            } catch (IOException e) {
+                var errMsg = "metrics=" + metricsOut.toString();
+                throw new AssertionError(errMsg, e);
+            }
+        }
+    }
+
+    // future-proof of accidental enum ordering change or extension
+    public void testEnsureMetricOrdinalsOrder() throws IOException {
```

Comment:
I opened https://github.com/elastic/elasticsearch/pull/111092 because I think we should have a more general-purpose test utility for this sort of thing.